scripts/src/java/org/oppia/android/scripts/wiki/WikiTableOfContentsCheck.kt

```diff
@@ -0,0 +1,100 @@
+package org.oppia.android.scripts.wiki
+
+import java.io.File
+
+/**
+ * Script for ensuring that the table of contents in each wiki page matches with its respective headers.
+ *
+ * Usage:
+ *   bazel run //scripts:wiki_table_of_contents_check -- <path_to_default_working_directory>
+ *
+ * Arguments:
+ * - path_to_default_working_directory: The default working directory on the runner for steps, and the default location of repository.
+ *
+ * Example:
+ *   bazel run //scripts:wiki_table_of_contents_check -- $(pwd)
+ */
+fun main(vararg args: String) {
+  // Path to the repo's wiki.
+  val wikiDirPath = "${args[0]}/wiki/"
+  val wikiDir = File(wikiDirPath)
+
+  // Check if the wiki directory exists.
+  if (wikiDir.exists() && wikiDir.isDirectory) {
+    processWikiDirectory(wikiDir)
+    println("WIKI TABLE OF CONTENTS CHECK PASSED")
+  } else {
+    println("No contents found in the Wiki directory.")
+  }
+}
+
+/**
+ * Checks every file in the wiki repo.
+ *
+ * @param wikiDir the default working directory
+ */
+fun processWikiDirectory(wikiDir: File) {
+  wikiDir.listFiles()?.forEach { file ->
+    checkTableOfContents(file)
+  }
+}
+
+/**
+ * Checks the contents of a single wiki file to ensure the accuracy of the Table of Contents.
+ *
+ * @param file the wiki file to process.
+ */
+fun checkTableOfContents(file: File) {
+  val fileContents = file.readLines()
+  val tocStartIdx = fileContents.indexOfFirst {
+    it.contains("## Table of Contents")
```

Comment:
What happens if there are multiple spaces after ##? do we want to handle that?

Reply:
Ah, I hadn't thought of that! I've now updated it to use a Regex with Ignore Case to also accept variations like 
- TABLE OF CONTENTS 
- Table of contents 
- Table Of   Contents
- etc.